core/modules/catalog-next/helpers/filterHelpers.ts

```diff
@@ -10,6 +10,7 @@ export const getSystemFilterNames: string[] = config.products.systemFilterNames
 export const changeFilterQuery = ({ currentQuery = {}, filterVariant }: {currentQuery?: any, filterVariant?: FilterVariant} = {}) => {
   const newQuery = JSON.parse(JSON.stringify(currentQuery))
   if (!filterVariant) return newQuery
+  if (newQuery['page']) delete newQuery['page']
```

Comment:
I would do it like that. It supports more cases
```js
if ('page' in newQuery && newQuery.page != 1) delete newQuery['page'];
```

Reply:
@Fifciu, when we are on page 1 of any category, the page-no is not present in the query, so I think there is no use in checking whether the `newQuery.page != 1` because if there is a page in the query then it is always not equal to 1.

Reply:
@Fifciu I applied the suggested changes, please review.